src/libraries/Microsoft.PowerFx.Core/Logging/Trackers/DelegationTelemetryInfo.cs

```diff
@@ -84,5 +85,33 @@ public static DelegationTelemetryInfo CreateImpureNodeTelemetryInfo(TexlNode nod
                     return new DelegationTelemetryInfo(node.ToString());
             }
         }
+
+        public static DelegationTelemetryInfo CreateAsyncNodeTelemetryInfo(TexlNode node, TexlBinding binding = null)
+        {
+            Contracts.AssertValue(node);
+            Contracts.AssertValueOrNull(binding);
+
+            switch (node.Kind)
+            {
+                case NodeKind.Call:
+                    var callNode = node.AsCall();
+                    var funcName = binding?.GetInfo(callNode)?.Function?.Name ?? string.Empty;
+                    return new DelegationTelemetryInfo(funcName);
+                default:
+                    return new DelegationTelemetryInfo(node.ToString());
```

Comment:
this is not safe for telemetry, node.tostring is a customer expression

Reply:
Changed this to `CreateEmptyDelegationTelemetryInfo` in the latest iteration